### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by yannikschaelte**
* add various tests for tables
* fix create_condition_df function

### CODE REVIEW

File: tests/test_conditions.py [hunk 1]
```diff
@@ -1,5 +1,9 @@
+import numpy as np
```
Comment (dweindl) [inline tests/test_conditions.py]: Codacy Issue found: [Missing module docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=4973405)

File: tests/test_observables.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    }).set_index(OBSERVABLE_ID)
+
+    with tempfile.NamedTemporaryFile(mode='w', delete=True) as fh:
+        file_name = fh.name
+        petab.write_observable_df(observable_df, file_name)
+        re_df = petab.get_observable_df(file_name)
+        assert (observable_df == re_df).all().all()
+
+
+def test_get_output_parameters():
```
Comment (dweindl) [inline tests/test_observables.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=4973405)

File: tests/test_conditions.py [hunk 3]
```diff
@@ -22,3 +27,69 @@ def test_get_parametric_overrides():
...
+    """Test conditions.get_condition_df."""
+    # condition df missing ids
+    condition_df = pd.DataFrame(data={
+        CONDITION_NAME: ['Condition 1', 'Condition 2'],
+        'fixedParameter1': [1.0, 2.0]
+    })
+
+    with tempfile.NamedTemporaryFile(mode='w', delete=False) as fh:
+        file_name = fh.name
+        condition_df.to_csv(fh, sep='\t', index=False)
```
Comment (dweindl) [inline tests/test_conditions.py]: use `write_condition_df`? makes it more future-proof... 
same for other tables and respective functions. some would requires an index.
not too passionate about it.

File: tests/test_observables.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    """Test measurements.get_output_parameters."""
+    # sbml model
+    document = libsbml.SBMLDocument(3, 1)
```
Comment (dweindl) [inline tests/test_observables.py]: use existing fixture `minimal_sbml_model`?

File: tests/test_petab.py [hunk 5]
```diff
@@ -366,3 +397,25 @@ def test_to_float_if_float():
     assert to_float_if_float("1e1") == 10.0
     assert to_float_if_float("abc") == "abc"
     assert to_float_if_float([]) == []
+
+
+def test_to_files(petab_problem):  # pylint: disable=W0621
+    """Test problem.to_files."""
+    sbml_file = tempfile.mkstemp()[1]
```
Comment (dweindl) [inline tests/test_petab.py]: tempfile.mkstemp:
> Unlike TemporaryFile(), the user of mkstemp() is responsible for deleting the temporary file when done with it.

Delete or use `tempfile.TemporaryDirectory`?

File: tests/test_conditions.py (hunk 3 above)
Comment (yannikschaelte) [reply tests/test_conditions.py]: there is a separate test for writing and reading below. for these tests here I wanted to be sure what is written to file.

File: tests/test_observables.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
+"""Tests related to petab.observables"""
```
Comment (dweindl) [inline tests/test_observables.py]: Codacy Issue found: [Similar lines in 2 files](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=4973405)

Comment (dweindl) [inline tests/test_observables.py]: Codacy Issue found: [Similar lines in 2 files](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=4973405)

**Pull request merged**